src/plaquette/pauli_frame.py

```diff
@@ -0,0 +1,371 @@
+def erase(pauli_frame: PauliFrame, p: float, qubit: int) -> tuple[PauliFrame, bool]:
+    r"""Erase the target qubit.
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        p: probability of erasing the qubit
+        qubit: target qubit
+    """
+    qubit_erased = plaquette.rng.random() < p
+    if qubit_erased:
+        # The qubit was erased, so we update the Pauli frame
+        pauli_frame = pauli_error_one_qubit(pauli_frame, 0.25, 0.25, 0.25, qubit)
+    return pauli_frame, qubit_erased
+
+
+def depolarize(pauli_frame: PauliFrame, targets: Sequence[int]) -> PauliFrame:
+    r"""Apply depolarization to each of the target qubits.
+
+    Applies an X, Y or Z Pauli error with 33% probability each.
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        targets: target qubits
+    """
+    for qubit in targets:
+        pauli_error_int = 1 + plaquette.rng.integers(0, 3)
+        pauli_frame = apply_pauli_from_int_sample(pauli_frame, pauli_error_int, qubit)
+    return pauli_frame
+
+
+def hadamard(pauli_frame: PauliFrame, targets: Sequence[int]) -> PauliFrame:
+    r"""Conjugate the frame with a Hadamard on the target qubits.
+
+    H transforms stabilizers according to :math:`Z \mapsto X` and
+    :math:`X \mapsto Z`
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        targets: 0-based target qubit index/indices.
+    """
+    num_qubits = get_num_qubits_from_frame(pauli_frame)
+    x_comp_idx = np.array(targets)
+    z_comp_idx = np.array(targets) + num_qubits
+
+    s = pauli_frame[x_comp_idx]
+    pauli_frame[x_comp_idx] = pauli_frame[z_comp_idx]
+    pauli_frame[z_comp_idx] = s
+    return pauli_frame
+
+
+def x(pauli_frame: PauliFrame, targets: Sequence[int]) -> PauliFrame:
+    """Conjugate the frame with a Pauli X on the target qubits.
+
+    This operation leaves the records in the frame unchanged:
+      * I: XIX = I
+      * X: XXX = X
+      * Y: XYX = -Y
+      * Z: XZX = -Z
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        targets: 0-based target qubit index/indices.
+    """
+    return pauli_frame
+
+
+def y(pauli_frame: PauliFrame, targets: Sequence[int]) -> PauliFrame:
+    """Conjugate the frame with a Pauli Y on the target qubits.
+
+    This operation leaves the records in the frame unchanged:
+      * I: YIY = I
+      * X: YXY = -X
+      * Y: YYY = Y
+      * Z: YZY = -Z
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        targets: 0-based target qubit index/indices.
+    """
+    return pauli_frame
+
+
+def z(pauli_frame: PauliFrame, targets: Sequence[int]) -> PauliFrame:
+    """Conjugate the frame with a Pauli Z on the target qubits.
+
+    This operation leaves the records in the frame unchanged:
+      * I: ZIZ = I
+      * X: ZXZ = -X
+      * Y: ZYZ = -Y
+      * Z: ZZZ = Z
+
+    Args:
+        pauli_frame: the Pauli frame to update
+        targets: 0-based target qubit index/indices.
+    """
+    return pauli_frame
+
+
+def cx(
```

Comment:
```suggestion
def controlled_not_gate(
```

This and the following changes are for consistency with how the pauli module is written.